### ISSUE TITLE
Treat build platform as a project setting

### ISSUE THREAD

**Issue by jspahrsummers** [Carthage/Carthage#356]
So it can be [specified in `Cartfile.private`](https://github.com/Carthage/Carthage/issues/262).

**Comment by kgn** [comment]
As mentioned in #877 I'd love a way to specify platforms in the `Cartfile`, something like:

```
platform iOS

github ...
```

**Comment by marcpalmer** [comment]
Initially I was into this but I'm not sure it makes sense. e.g. if you have a project with cross-platform outputs, you really need a single Cartfile that does specify which platforms to build each dependency for.

An example would be an app that has iOS and tvOS versions, and the iOS version has features that are not available in tvOS - and it does this via some dependencies. You would want to build everything for iOS + tvOS, *except* those frameworks that are only desired on iOS. A single Cartfile would not enable this so the "platforms" would need to be specified per dependency.

Otherwise, you'd have no choice but to have separate project folders and separate Cartfile even though these are just variants of the same app?

**Comment by jspahrsummers** [comment]
If implemented, this setting belongs in `Cartfile.private` (as mentioned in the issue body), so it wouldn't apply transitively, but instead be a "local" project configuration. Most useful for applications.

**Comment by mlilback** [comment]
Also useful for those of us who are only building a Mac version. It can be defaulted to all, so there would be no backward compatibility.

**Comment by Cramsden** [comment]
MERGE PLZ!!!!

**Comment by craigotis** [comment]
This would be very great to have. With an iOS app depending on ~10 different libraries, I could save a lot of time not having to build the dependencies for AppleTV, the Apple Watch, and the Mac.

**Comment by jspahrsummers** [comment]
@craigotis You can already provide `--platform` to specify that. This is just about being able to set defaults.

**Comment by Dschee** [comment]
It would still be very nice to see this feature implemented, is there any work done on this already?

**Comment by mdiep** [comment]
No. This is waiting on a more flexible format for Cartfiles. But we're also focused on cleaning up and improving existing code/functionality.

**Comment by Dschee** [comment]
Is there an issue of the discussion about a more flexible Cartfile so we can keep staying updated?

**Comment by joeblau** [comment]
@jspahrsummers You keep mentioning this:

> You can already provide `--platform` to specify that.

But there is no way to default your platform so that you don't have to type it in locally if you're building.  I think that's what people are asking for.

**Comment by marcvanolmen** [comment]
Being able to specify platform in some kind of form in the cartfile is really important for example we use different frameworks for iOS and tvOS and now we have extra frameworks being build on both platforms.

**Comment by CodeEagle** [comment]
## Support Configuration in `Cartfile.private`
> In my fork [CodeEagle/Carthage](https://github.com/CodeEagle/Carthage).
## Support  [`Punic`](https://github.com/schwa/punic) Features
- override 
- skip
## Support  [`RocketBoot` ](https://github.com/CodeEagle/RocketBoot)
> Speed up App Launching
